src/main/kotlin/org/rust/cargo/project/CargoProjectOpenProcessor.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupManager
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.platform.PlatformProjectOpenProcessor
+import com.intellij.projectImport.ProjectOpenProcessor
+import org.rust.cargo.CargoConstants
+import org.rust.cargo.icons.CargoIcons
+import org.rust.cargo.project.model.guessAndSetupRustProject
+import javax.swing.Icon
+
+class CargoProjectOpenProcessor : ProjectOpenProcessor() {
+    override fun getIcon(): Icon? = CargoIcons.ICON
+    override fun getName(): String = "Cargo"
+
+    override fun canOpenProject(file: VirtualFile?): Boolean = isCargoToml(file)
+    private fun isCargoToml(file: VirtualFile?) = FileUtil.namesEqual(file?.name, CargoConstants.MANIFEST_FILE)
```

Comment:
Maybe inline this method? Currently, we use it only in one place

Reply:
But if you think that it really simplify code, let's specify return type of `isCargoToml` explicitly

Reply:
inlined